src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/ViewComponentTagHelperChunkVisitor.cs

```diff
@@ -0,0 +1,195 @@
+
+        protected override void Visit(ParentChunk parentChunk)
+        {
+            foreach (var chunk in parentChunk.Children)
+            {
+                Accept(chunk);
+            }
+        }
+
+        protected override void Visit(TagHelperChunk chunk)
+        {
+            foreach (var descriptor in chunk.Descriptors)
+            {
+                string shortName;
+                descriptor.PropertyBag.TryGetValue(
+                    ViewComponentTagHelperDescriptorConventions.ViewComponentNameKey,
+                    out shortName);
+
+                if (shortName != null)
+                {
+                    var typeName = $"__Generated__{shortName}ViewComponentTagHelper";
+
+                    if (_writtenViewComponents.Add(typeName))
+                    {
+                        WriteClass(descriptor);
+                    }
+                }
+            }
+        }
+
+        private void WriteClass(TagHelperDescriptor descriptor)
+        {
+            // Add target element.
+            BuildTargetElementString(descriptor);
+
+            // Initialize declaration.
+            var tagHelperTypeName = typeof(TagHelper).FullName;
+
+            var shortName = descriptor.PropertyBag[ViewComponentTagHelperDescriptorConventions.ViewComponentNameKey];
+            var className = $"__Generated__{shortName}ViewComponentTagHelper";
+
+            using (Writer.BuildClassDeclaration("public", className, new[] { tagHelperTypeName }))
+            {
+                // Add view component helper.
+                Writer.WriteVariableDeclaration(
+                    $"private readonly {_context.IViewComponentHelperTypeName}",
+                    ViewComponentTagHelperVariable,
+                    value: null);
+
+                // Add constructor.
+                BuildConstructorString(className);
+
+                // Add attributes.
+                BuildAttributeDeclarations(descriptor);
+
+                // Add process method.
+                BuildProcessMethodString(descriptor);
+            }
+        }
+
+        private void BuildConstructorString(string className)
+        {
+            var viewComponentHelperVariable = "viewComponentHelper";
+
+            var helperPair = new KeyValuePair<string, string>(
+                _context.IViewComponentHelperTypeName,
+                viewComponentHelperVariable);
+
+            using (Writer.BuildConstructor( "public", className, new[] { helperPair }))
+            {
+                Writer.WriteStartAssignment(ViewComponentTagHelperVariable)
+                    .Write(viewComponentHelperVariable)
+                    .WriteLine(";");
+            }
+        }
+
+        private void BuildAttributeDeclarations(TagHelperDescriptor descriptor)
+        {
+            Writer.Write("[")
+              .Write(typeof(HtmlAttributeNotBoundAttribute).FullName)
+              .WriteParameterSeparator()
+              .Write(_context.ViewContextTypeName)
```

Comment:
Probably makes sense. That'd be more consistent with how the rest of the Razor system works.